spec.md

```diff
@@ -266,15 +266,10 @@ message GCSParameters {
 }
 
 message Protocol {
-    // ProtocolName is the name of the protocol
-    ProtocolName name = 1;
-    // version is the name of the protocol version
-    string version = 2;
-
     oneof type {
-        S3Parameters s3 = 3;
-        AzureBlobParameters azureBlob = 4;
-        GCSParameters gcs = 5;
+        S3Parameters s3 = 1;
+        AzureBlobParameters azureBlob = 2;
+        GCSParameters gcs = 3;
```

Comment:
nit: why suffix these with `Parameters`?

Reply:
Protocol name S3 otherwise collides with the Parameter type S3

Reply:
Do you mean it collides with `enum ProtocolName`? if so, is that enum still needed (it is removed in this PR)?